src/main/java/net/minecraftforge/common/ForgeHooks.java

```diff
@@ -976,6 +976,10 @@ public static CriticalHitEvent getCriticalHit(Player player, Entity target, bool
         return null;
     }
 
+    /**
+     * @deprecated See {@linkplain ForgeEventFactory}
+     */
+    @Deprecated //todo: remove and use ForgeEventFactory.onAdvancementEarned instead in 1.19
```

Comment:
Mark deprecations for removal as such in the annotation:
```suggestion
    @Deprecated(forRemoval = true, since = "1.18.1") // TODO: remove and use ForgeEventFactory.onAdvancementEarned instead in 1.19
```